src/core/src/core_logic/PatchInstaller.py

```diff
@@ -275,21 +275,50 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         stopwatch_for_batch_install_process = Stopwatch(self.env_layer, self.telemetry_writer, self.composite_logger)
         stopwatch_for_batch_install_process.start()
 
-        # Check for packages available and install them in batches. Some packages may not be installed due to:
-        # (a) Not enough time remaining in maintenance window
-        # (b) Failure during package installation
-        installed_update_count, patch_installation_successful, maintenance_window_batch_cutoff_reached, packages, package_versions = self.install_packages_in_batches(
-            all_packages, all_package_versions, packages, package_versions, maintenance_window, package_manager)
+        total_packages_to_install_count = len(packages)
+        maintenance_window_batch_cutoff_reached = False
```

Comment:
This should always be a check, where we validate time remaining, instead of assuming there will be time left since actual package installation hasn't started yet.

Reason for this is, patch operations are often not completed in a single occurence, due to restarts, machine reboots (by an external entity), etc. There could be an instance where maintenance window has exceeded at this point in code because it is a returning operation at this line.

Reply:
nvm, thought this was the overall maintenance window for installation. Closing this comment